added rh-rpm-advisories pipelines

rashiddm opened the pull request:
## Describe your changes

## Relevant Jira
https://issues.redhat.com/browse/ROK-1303

## Checklist before requesting a review
- [x] I have marked as draft or added `do not merge` label if there's a dependency PR
  - If you want reviews on your draft PR, you can add reviewers or add the `release-service-maintainers` handle if you are unsure who to tag
- [x] My commit message includes `Signed-off-by: My name <email>`
- [x] I read CONTRIBUTING.MD and [commit formatting](CONTRIBUTING.md#commit-message-formatting-and-standards)
- [x] I have run the README.md generator script in `.github/scripts/readme_generator.sh` and verified the results using `.github/scripts/check_readme.sh`

xiangge commented on pipelines/managed/rh-rpm-advisories/rh-rpm-advisories.yaml:
```diff
@@ -0,0 +1,273 @@
+              {"resultIndex": 1, "key": ".conforma.timeout", "default": "8h0m0s"},
+              {"resultIndex": 2, "key": ".sign.cosignSecretName", "default": ""},
+              {"resultIndex": 3, "key": ".pyxis.secret"},
+              {"resultIndex": 4, "key": ".pyxis.server", "default": "production"},
```
```suggestion
```
We won't have pyxis data on RPM I believe.

xiangge commented on pipelines/managed/rh-rpm-advisories/rh-rpm-advisories.yaml:
```diff
@@ -0,0 +1,273 @@
+              #!/bin/bash
+              echo "create-advisory"
+              echo -n "" > $(results.advisory_url.path)
+              echo -n "" > $(results.advisory_internal_url.path)
```
@rashiddm Please add the finally task here, e.g. https://github.com/konflux-ci/release-service-catalog/blob/7b9746ccf1a68dce36492aa042398cdb0e990243/pipelines/managed/rh-advisories/rh-advisories.yaml#L1098

rashiddm commented:
@xiangge  I have updated the PR as per feedback.

xiangge commented on pipelines/managed/rh-rpm-advisories/README.md:
```diff
@@ -0,0 +1,52 @@
+3. **collect-task-params** - Extracts task parameters from the collected data
+4. **reduce-snapshot** - Reduces the snapshot for SingleComponentMode support
+5. **verify-conforma** - Verifies the release using Conforma (Enterprise Contract)
+6. **create-advisory** - Creates an advisory for the release (currently a placeholder that prints "create-advisory")
```
Need to mention cleanup-internal-requests

xiangge commented:
LGTM besides on comment on the readme.

scoheb commented on pipelines/managed/rh-rpm-advisories/rh-rpm-advisories.yaml:
```diff
@@ -0,0 +1,285 @@
+apiVersion: tekton.dev/v1
+kind: Pipeline
+metadata:
+  name: rh-rmp-advisories
```
spelling mistake

rashiddm replied:
okay let me fix this.

scoheb commented on pipelines/managed/rh-rpm-advisories/rh-rpm-advisories.yaml:
```diff
@@ -0,0 +1,285 @@
+    tekton.dev/tags: release
+spec:
+  description: |-
+    Tekton pipeline to release content to registry.redhat.io registry and create an advisory.
```
needs updating

rashiddm replied:
can you please explain this, what changes needed here ?

scoheb replied:
the description does not makes sense anymore. you are not going to be pushing to a registry since you are dealing with rpms

rashiddm replied:
```
Tekton pipeline to release RPM content and create an advisory.
    This is the first version of the RPM advisory pipeline with basic tasks for
    verifying access, collecting data, reducing snapshot, verifying with Conforma,
    and creating an advisory.
```
    
  is this okay to put there inplace of descriptions, do let me know
  cc @xiangge @scoheb

xiangge replied:
Tekton pipeline to release RPM content.
This is the first version of the RPM release pipeline with basic tasks for
verifying access, collecting data, reducing snapshot, verifying with Conforma,
and mock task of creating advisory.

See my suggestion on the readme.

rashiddm replied:
got it thanks.

xiangge commented on pipelines/managed/rh-rpm-advisories/README.md:
```diff
@@ -0,0 +1,53 @@
+This is the first version of the RPM advisory pipeline with basic tasks for
+verifying access, collecting data, reducing snapshot, verifying with Conforma,
+and creating an advisory.
+
```
```suggestion
Tekton pipeline to release RPM content.
This is the first version of the RPM release pipeline with basic tasks for
verifying access, collecting data, reducing snapshot, verifying with Conforma,
and mock task of creating advisory.

```

swickersh commented:
/review

rashiddm committed:
feat: create-advisory-pipeline

This is the first version of the RPM release pipeline with
basic tasks for verifying access, collecting data, reducing
snapshot, verifying with Conforma, and mock task of creating
advisory.

Signed-off-by: Md Rashid <mmdrashi@redhat.com>

rashiddm commented:
@mmalina  @xiangge  fixed readme and shell issues, can you please review and run ci checks again.
